ssh-key/src/mpint.rs

```diff
@@ -114,6 +115,12 @@ impl PartialEq for Mpint {
     }
 }
 
+impl Hash for Mpint {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        self.inner.hash(state)
+    }
+}
```

Comment:
Why not a derived impl?

Reply:
Clippy is of the opinion that when PartialEq is implemented manually, Hash should as well: https://rust-lang.github.io/rust-clippy/master/index.html#derived_hash_with_manual_eq

Reply:
Hmm, it is right about one thing: the `PartialEq` impl is customized to compare `Mpint`s in constant time since they may contain secrets.

A `Hash` impl directly on the type is a bit problematic since it could potentially leak secret information.

Alternatively instead of handwriting this impl so it can be derived on the public key types, you could handwrite those impls so `Mpint` isn't accidentally `Hash`ed in secret contexts.